### ISSUE TITLE
Is this line valid?

### ISSUE THREAD

**Issue by hellowin** [peter-kish/gloot#101]
https://github.com/peter-kish/gloot/blob/7f86281808b230d20f92ae6b84944df4b7dfa46e/addons/gloot/inventory_item.gd#L30-L31

Is this line have a possibility of getting triggered? Because `old_prototype_id` is set as `prototype_id`, in line 31 there's an `if` statement for that, when is that `if` expected to be called?

**Comment by peter-kish** [comment]
Yup, the whole function looks super wrong. I probably did some refactoring at some point and messed it up.
What I wanted to do is probably something like this:
```gd
if new_prototype_id == prototype_id:
    return
reset_properties()
prototype_id = new_prototype_id
update_configuration_warnings()
emit_signal("prototype_id_changed")
```
I will do some testing with this code above and then push a fix asap.

**Comment by peter-kish** [comment]
Should be fixed in 535945b70bb94fa03bc0fa1f90fed9a61346f3ed
